plugins/woocommerce/src/Internal/Admin/BlockTemplates/AbstractBlock.php

```diff
@@ -161,38 +161,36 @@ public function set_attributes( array $attributes ) {
 
 	/**
 	 * Get the template that this block belongs to.
-	 *
-	 * @throws \RuntimeException If the block does not have a root template.
 	 */
 	public function &get_root_template(): BlockTemplateInterface {
-		if ( is_null( $this->root_template ) ) {
-			throw new \RuntimeException( 'The block does not have a root template.' );
-		}
-
 		return $this->root_template;
 	}
 
 	/**
 	 * Get the parent block container.
-	 *
-	 * @throws \RuntimeException If the block does not have a parent.
 	 */
 	public function &get_parent(): ContainerInterface {
-		if ( is_null( $this->parent ) ) {
-			throw new \RuntimeException( 'The block does not have a parent.' );
-		}
-
 		return $this->parent;
 	}
 
 	/**
-	 * Detach the block from its parent block container and the template it belongs to.
+	 * Remove the block from its parent.
 	 */
-	public function detach() {
-		$this->parent        = null;
-		$this->root_template = null;
+	public function remove() {
+		$this->parent->remove_block( $this->id );
 	}
 
+	/**
+	 * Check if the block is detached from its parent block container or the template it belongs to.
+	 *
+	 * @return bool True if the block is detached from its parent block container or the template it belongs to.
+	 */
+	public function is_detached(): bool {
+		$is_in_parent        = $this->parent->get_block( $this->id ) === $this;
+		$is_in_root_template = $this->get_root_template()->get_block( $this->id ) === $this;
+
+		return ! $is_in_parent || ! $is_in_root_template;
```

Comment:
Nitpick: This could be simplified by changing to ` !(a && b)`. I would find it a little bit easier to read.

```php
return ! ( $is_in_parent && $is_in_root_template ) ;
```

Reply:
Done!